### PR TITLE
fix: acme: invert check for valid (identifier, challenge type) pairs [WPB-10103]

### PR DESCRIPTION
**Pull request by istankovic**
Explicitly check for the cases we allow, and let others result in an error.

### CODE REVIEW

File: acme/src/authz.rs [hunk 1]
```diff
@@ -93,39 +93,37 @@ impl AcmeAuthz {
...
-        let token = base64::prelude::BASE64_URL_SAFE_NO_PAD
-            .decode(&challenge.token)
-            .map_err(|_| AcmeAuthzError::InvalidBase64Token)?;
-
-        // token have enough entropy (at least 16 bytes)
-        // see https://datatracker.ietf.org/doc/html/rfc8555#section-11.3
-        const RECOMMENDED_TOKEN_ENTROPY: usize = 128 / 8;
-        if token.len() < RECOMMENDED_TOKEN_ENTROPY {
-            return Err(AcmeAuthzError::InvalidTokenEntropy.into());
+        match (&self.identifier, challenge.typ) {
```
Comment (augustocdias) [inline acme/src/authz.rs:96]: I would prefer to use `if matches!` since it is just one case here...

File: acme/src/authz.rs [hunk 2]
```diff
@@ -93,39 +93,38 @@ impl AcmeAuthz {
...
+            (AcmeIdentifier::WireappDevice(_), AcmeChallengeType::WireDpop01)
+                | (AcmeIdentifier::WireappUser(_), AcmeChallengeType::WireOidc01)
+        ) {
+            let now = time::OffsetDateTime::now_utc().unix_timestamp();
+
+            let is_expired = self
+                .expires
+                .map(time::OffsetDateTime::unix_timestamp)
+                .map(|expires| expires < now)
+                .unwrap_or_default();
```
Comment (augustocdias) [inline acme/src/authz.rs:107]:
```suggestion
                .is_some_and(|expires| expires < now);
```

**Review by augustocdias — APPROVED** [review]

File: acme/src/authz.rs (hunk 2 above)
Comment (augustocdias) [reply acme/src/authz.rs:107]: this is minor... feel free to just ignore it if you want

Comment (istankovic) [reply acme/src/authz.rs:107]: Actually, I quite like your `is_some_and` suggestion, reads much more nicely. Thanks. :+1:

**Pull request merged**